Automattic/wp-calypso
# Purchases: Display placeholders on `ManagePurchase` if visiting after data has changed

`ManagePurchase` was updated to display placeholders whenever the purchase data is being fetched. We fetch this data whenever we mount `ManagePurchase`, so we end up displaying the placeholders even if the client has the data it needs to display the purchase.

This commit updates `ManagePurchase` to only display the placeholders if it is the destination after a renewal or card data has changed.

**Testing**
We need to check that the data on `ManagePurchase` fetches when editing payment details and renewing.

*Editing payment details*
- Visit `/purchases`
- When purchases have loaded, click on a purchase
- Assert that no placeholders are displayed after you click on a purchase
- Visit `/purchases/:site/:id` for a purchase made with a credit card.
- Click 'Edit Payment Method'
- Change the credit card used the payment.
- Assert that you are redirected to the `ManagePurchase` page for the given purchase with a success notice, and that the placeholders are displayed.

*Renewing*
- Visit `/purchase/:site` for a subscription that expires within 30 days.
- Click the button to renew the purchase and complete checkout.
- Assert that you are redirected to the `ManagePurchase` page for the given purchase with a success notice, and that the placeholders are displayed.


cc @scruffian
